prowspecs/specs.go

```diff
@@ -36,7 +36,7 @@ var knownBranches map[string]BranchSpec = map[string]BranchSpec{
 			Branch: "release-1.10",
 
 			// Freeze test images used.
-			Image: "eu.gcr.io/jetstack-build-infra-images/bazelbuild:20220830-c65cd19-4.2.1",
+			Image: "eu.gcr.io/jetstack-build-infra-images/make-dind:20230406-0ef4440-bullseye",
```

Comment:
suggestion: Having older releases have their images be "pinned" was an intentional choice in #90  - does this not defeat the point of that PR?

IMO either we should decide to use CommonTestImage everywhere or leave the older images alone. I'd be happy to use CommonTestImage everywhere, but that was explicitly rejected in #90

Reply:
The images were changed from bazelbuild to make-dind in https://github.com/jetstack/testing/pull/809.
The other follow-up PRs just upgraded bazelbuild. Whether we want to keep doing this automatically, I think we have to discuss (eg. the images are still "pinned" to bullseye).

Reply:
After discussion in standup it seems clear that we want to use `CommonTestImage` for every Image parameter until we have a reason not to :+1: